packages/expo-cli/src/accounts.ts

```diff
@@ -95,7 +110,171 @@ export async function login(options: CommandOptions): Promise<User> {
   }
 }
 
-async function _usernamePasswordAuth(username?: string, password?: string): Promise<User> {
+/**
+ * Prompt for an OTP with the option to cancel the question by answering empty (pressing return key).
+ */
+async function _promptForOTPAsync(cancelBehavior: 'cancel' | 'menu'): Promise<string | null> {
+  const enterMessage =
+    cancelBehavior === 'cancel' ? 'press enter to cancel' : 'press enter for other options';
+  const otpQuestion: Question = {
+    type: 'input',
+    name: 'otp',
+    message: `One-time Password or Backup Code (${enterMessage}):`,
+  };
+
+  const { otp } = await prompt(otpQuestion);
+  if (!otp || otp === '') {
+    return null;
+  }
+
+  return otp;
+}
+
+/**
+ * Prompt for user to choose a backup OTP method. If selected method is SMS, a request
+ * for a new  will be sent to that method. Then, prompt for the OTP, and retry the user login.
+ */
+async function _promptForBackupOTPAsync(
+  username: string,
+  password: string,
+  secondFactorDevices: SecondFactorDevice[]
+): Promise<string | null> {
+  const nonPrimarySecondFactorDevices = secondFactorDevices.filter(device => !device.is_primary);
+
+  if (nonPrimarySecondFactorDevices.length === 0) {
+    throw new Error('No other second factor devices set up for user');
```

Comment:
Since these error messages are user-facing, try to phrase this to be more directed at the user e.g.
```suggestion
    throw new Error('There are no other second-factor devices set up for you');
```
```suggestion
    throw new Error('No other second factor devices set up for user');
```